Added more explanation of apps in intro, more specifics in applicatio…

ekansa opened the pull request:
…n section

### brief description of changes
This is just a start to get some feedback, but is not at all ready for merging.

#### issues addressed
Our conversation about Arches applications and how to explain their significance in developer guidance.

#### further comments
Please review and consider with @apeters 's proposed forum post. I think I'm covering some of the points, but may be garbling some of this. It would be great to provide some specifics about how to make a new Arches Application (starting from a project, right?). I read about how Django does it but the terminology seems to differ from Arches terminology a little. Anyway, those terminology issues should be an important topic for review.

---

This box **must** be checked
- [ ] the PR branch was originally made from the base branch

This box **should** be checked
- [ ] after these changes the docs build locally without error

This box **should only** be checked you intend to follow through on it (we can do it on our end too)
- [ ] I will `cherry-pick` all commits in this PR into other branches that should have them _after_ this PR is merged

apeters commented on docs/developing/getting-started/customization-considerations.rst:
```diff
@@ -7,8 +7,19 @@ If you are leading a project or organization considering customizing Arches soft
 
-More Sustainable Pathways toward Customization
-==============================================
+
+Customized Applications
+=======================
+While the Arches extensions architecture offers a great deal of flexibility (see below), there may be scenarios where you need additional flexibility. From a sustainability and maintenance perspective, this scenario has important risks that need to be understood and factored into long term resource planning and engineering.
+
+Managing long term sustainability and maintenance risks should be a core software engineering focus. As much as possible, you should ideally isolate your customized module as much as possible from the core of Arches. One way preferred way to accomplish this is to develop  **Arches Applications** (see: :ref:`Creating Applications`), which are discrete Python packages that can be integrated into on ...
+
+Arches Applications are a powerful way to create customizations that are isolated from the core of Arches. This isolation can help to reduce the risk of future maintenance and upgrade challenges. Arches Applications can be developed and maintained independently of the core Arches software, and can be shared with the wider Arches community. This can help to ensure that your customizations are more ...
+
+
```
```suggestion
```

apeters commented on docs/developing/getting-started/customization-considerations.rst:
```diff
@@ -27,13 +38,6 @@ To increase the likelihood that customizations will have long term compatibility
 Adherence to the extension design patterns helps to isolate your customizations from changes to the core of Arches. Following Arches extensions design patterns will also increase the likelihood that there will be relevant documentation and community help if the extensions need updates in the future. Certain customizations are easier to maintain over time. For example, an overwritten HTML template ...
 
 
```
```suggestion

Arches Applications
=======================
While the Arches extensions (see above) architecture offers a great deal of flexibility, there may be scenarios where you need additional flexibility. From a sustainability and maintenance perspective, ideally you should isolate your customized module as much as possible from the core of Arches. One way to accomplish this is to develop an  **Arches Application** (see: :ref:`Creating Applications`), which are discrete Python packages that can be integrated into one or more Arches projects. The **Arches Applications** documentation details their development as well as their sustainability advantages.

Arches Applications are a powerful way to create customizations that are isolated from the core of Arches. This isolation can help to reduce the risk of future maintenance and upgrade challenges. Arches Applications create reusable components that can be developed and maintained independently of the core Arches software, and can be shared with the wider Arches community. This can help to ensure that your customizations are more maintainable over time and can help to reduce the overall cost of development and maintenance.
```

apeters commented on docs/developing/getting-started/customization-considerations.rst:
```diff
@@ -7,8 +7,19 @@ If you are leading a project or organization considering customizing Arches soft
+
+
+
+Customization of Simple Extensions
```
```suggestion
Custom Extensions
```

ekansa commented:
@apeters I got back to this and incorporated your changes, and fixed a few style issues. This is temporarily viewable here: https://arches.readthedocs.io/en/applications_7_6/developing/extending/creating-apps/

ekansa commented:
Merging this, because it's better than what we already had! We can return to this to make additional improvements.
